File: src/main/java/com/logviewer/ui/component/tailView.js
com_logviewer_ui_component_TailView = function() {

	var e = this.getElement();
	var scroll = false;

	e.innerHTML = "<pre></pre>";

	this.onStateChange = function() {
		scroll = this.getState().scroll;

		if (this.getState().value){
			this.add(this.getState().value);
		} else {
			this.setText(this.getState().text);
		}
	}

	this.setText = function (value){
		e.getElementsByTagName("pre")[0].innerHTML = value;
		this.scroll();
	}

	this.add = function (value){
	    e.getElementsByTagName("pre")[0].innerHTML += value;
	    this.scroll();
	}

	this.clear = function (value){
        e.getElementsByTagName("pre")[0].innerHTML = '';
    }

	this.scroll = function(){
		 if (scroll == true) {
			window.setInterval(function() {
			  e.parentNode.scrollTop = e.parentNode.scrollHeight;
			}, 5000);
		}
	}

}

File: src/main/java/com/logviewer/ui/component/TailView.java
package com.logviewer.ui.component;

/**
 * Created by 953682 on 20/04/2015.
 */
@SuppressWarnings("serial")
@com.vaadin.annotations.JavaScript({"tailView.js"})
public class TailView extends com.vaadin.ui.AbstractJavaScriptComponent {

    private int countLines = 0;

    public TailView(final String text) {
        add(text);
    }

    @Override
    protected TailViewState getState() {
        return (TailViewState) super.getState();
    }

    public void setText(String text){
        getState().text = text;
        getState().value = null;
    }

    public void add(String value){
        getState().value = value;
        countLines++;
    }

    public void clear(){
        countLines = 0;
        setText("");
    }

    public void setScroll(boolean scroll){
        getState().scroll = scroll;
    }

    public int getCountLines() {
        return countLines;
    }

    public void setCountLines(int countLines) {
        this.countLines = countLines;
    }
}
